tests/final_chain_test.cpp

```diff
@@ -277,6 +278,36 @@ TEST_F(FinalChainTest, nonce_skipping) {
   advance({std::make_shared<Transaction>(5, 13, 0, TRX_GAS, dev::bytes(), key.secret(), key.address())});
 }
 
+TEST_F(FinalChainTest, initial_validators) {
+  const dev::KeyPair key = dev::KeyPair::create();
+  const std::vector<dev::KeyPair> validator_keys = {dev::KeyPair::create(), dev::KeyPair::create(),
+                                                    dev::KeyPair::create()};
+  cfg.state.genesis_balances = {};
+  cfg.state.dpos.emplace();
+  cfg.state.dpos->eligibility_balance_threshold = 100000;
+  cfg.state.dpos->vote_eligibility_balance_step = 10000;
+  cfg.state.dpos->maximum_stake = 10000000;
+  cfg.state.dpos->minimum_deposit = 1000;
+  cfg.state.dpos->eligibility_balance_threshold = 1000;
+
+  for (const auto& vk : validator_keys) {
+    state_api::ValidatorInfo validator{vk.address(), key.address(), 0, "", "", {}};
+    validator.delegations.emplace(key.address(), cfg.state.dpos->maximum_stake);
+    validator.delegations.emplace(validator_keys.front().address(), cfg.state.dpos->maximum_stake);
```

Comment:
For the same validator, you are creating 2 delegators with  maximum_stake delegation, right ? If that is the case, the second delegation should not get through because maximum_stake is max. total stake for validator...

Reply:
maybe we should rename maximum_stake to something like `validator_max_total_stake` so it is not so confusing ?